ddtrace/internal/_encoding.pyx

```diff
@@ -314,3 +308,48 @@ cdef class MsgpackEncoder(object):
             return struct.pack(">BH", 0xdc, count) + buf
         else:
             return struct.pack(">BI", 0xdd, count) + buf
+
+
+cdef class _StringTable(list):
+    cdef dict _index
+
+    def __init__(self):
+        super(_StringTable, self).__init__([""])
+        self._index = {"": 0}
+
+    def index(self, string):
+        cdef int index
+
+        if not string:
+            return 0
+        
+        index = self._index.get(string, 0)
+        if not index:
+            index = len(self)
+            self.append(string)
+            self._index[string] = index
```

Comment:
We don't need a lock around this do we?

Reply:
If that's compatible with https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/profiling/exporter/pprof.pyx#L51-L68 could we share the code here? 🙏